### ISSUE TITLE
v1 checklist

### ISSUE THREAD

**Issue by cmcaine** [cmcaine/tridactyl#24]
Remaining work before first public release:

- [x]  - improve exmode parser with data from macros (cmd_params map)
- [x]  - link hinting
- [ ]  - find mode (/nN)
- [x]  - (*shit workaround*) preventDefault on mapped keys (generalise normal mode parser into key sequence parser and send maps to keyboard shim)
- [x]  - basic custom maps
- [x]  - persistent config (just maps for now)
- [x]  - don't accept untrusted keyevents

Other important work:

 - Completion system (Koushien's working on this :)
 - Documentation generation and help function
 - Better maps
 - `:set` (configuration)
 - error bar (and error propagation from everywhere to that bar)
 - nativeMessenger (for editing userChrome.css, rc files, opening file:// URIs, etc)
 - `:help` (#23)
 - Move clipboard read into background

Lower priority:

 - Other missing ecmds from vimperator
 - JS Eval/custom excmds
 - autocmds
 - mode notification

**Comment by P4Cu** [comment]
Putting this here as we may find some of these usable later:
https://chudq.wordpress.com/2009/01/24/my-favourite-vimperator-keys/
http://sheet.shiar.nl/vimperator

my favourite is `gi` which is mandatory to be able to chat ;-)

**Comment by bovine3dom** [comment]
Firefox's quick find mode works again now. I suggest we just bind n to "to find next occurence press Ctrl+G" instead?

Blocked by "provide feedback / errors to users".

**Comment by cmcaine** [comment]
Signed xpi (with extension changed to zip to make github shut up).

[tridactyl-1.0.4-an+fx.zip](https://github.com/cmcaine/tridactyl/files/1456663/tridactyl-1.0.4-an.fx.zip)

**Comment by cmcaine** [comment]
Next biggest user-impact features are probably search, error messages, better completion, command abbreviation

**Comment by cmcaine** [comment]
Up to date releases are here:

https://addons.mozilla.org/en-US/firefox/addon/tridactyl-vim/versions/beta

**Comment by bovine3dom** [comment]
I'm going to close this because I think we released v1.0.0 a while ago...
